sqlmesh/core/context.py

```diff
@@ -308,14 +308,16 @@ def scheduler(self, environment: t.Optional[str] = None) -> Scheduler:
             console=self.console,
         )
 
+    def _new_state_sync(self) -> StateSync:
```

Comment:
[Nit] Can we move this to the bottom so that public API is at the top of the class defintiion